eclipse-theia/theia
# Add a title for custom title bar style

#### What it does

Closes https://github.com/eclipse-theia/theia/issues/7076

When selecting the `custom` title bar style, the application currently shows no title. This change adresses the issue by trying to replicate what VSCode does with its title bars. Also aligns the title behavior of Theia to VSCode which shows the currently opened workspace and active main area widget in the title. For this, some more title-based infrastructure has been introduced to facilitate different contributions to the title.

#### How to test

1. Start the electron app on Windows or Linux and select the `custom` title bar style in the preferences
2. Assert that the title bar behaves like VSCode when resizing the window.
3. Open a workspace and start opening files or other widgets which appear in the main area. The currently opened workspace and the currently selected main area widget appear in the title bar.
4. Closing all workspaces/main area widgets removes these contributions from the title bar.
5. Assert that these changes also work correctly when using the `native` title bar style and in the browser.

#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
